llvm/lib/MC/ELFObjectWriter.cpp

```diff
@@ -725,7 +725,13 @@ void ELFWriter::computeSymbolTable(
         HasLargeSectionIndex = true;
     }
 
+    // Temporary symbols generated for certain assembler features (.eh_frame,
+    // .debug_line) of an empty name may be referenced by relocations due to
+    // linker relaxation. Rename them to ".L0" to match the gas fake label name
+    // and allow ld/objcopy --discard-locals to discard such symbols.
     StringRef Name = Symbol.getName();
+    if (Name.empty())
+      Name = ".L0";
```

Comment:
It's ".L0 " rather than ".L0" in binutils actually

https://github.com/bminor/binutils-gdb/blob/master/include/opcode/riscv.h#L381

Ref: https://github.com/bminor/binutils-gdb/commit/884b49e3a91504b9c5528343823d67f4419b73bb

Reply:
Oh, interesting point. If these symbols are "discarded after linking by default", is that `symbol_is_valid` filtering relevant only when the defaults are overridden and they are not being discarded?

Reply:
Thanks for pointing this out. I invoked `readelf -x .strtab a.o` but did not read the output carefully to notice the space.
Switched to `.L0 `

Reply:
I assume that fake labels `.L0\x01` for other ports cannot be retained in .symtab . Even if they do, LLVM likely doesn't want to customize on this.